go-controller/pkg/factory/factory.go

```diff
@@ -204,6 +211,19 @@ func NewMasterWatchFactory(ovnClientset *util.OVNMasterClientset) (*WatchFactory
 			return nil, err
 		}
 	}
+
+	if util.IsNetworkSegmentationSupportEnabled() {
+		if err := userdefinednetworkapi.AddToScheme(userdefinednetworkscheme.Scheme); err != nil {
+			return nil, err
+		}
+
+		wf.udnFactory = userdefinednetworkapiinformerfactory.NewSharedInformerFactory(ovnClientset.UserDefinedNetworkClient, resyncInterval)
```

Comment:
From what I understand, the UDN controller should run on ovn-k control plane, it can either be the ovnkube-cluster-manager or ovnkube-master pod, depending on the configuration decided here: 
https://github.com/ovn-org/ovn-kubernetes/blob/61b0a6bd59ba6f506f1277e2b4fa794c26dcd22d/go-controller/cmd/ovnkube/ovnkube.go#L585